### PR TITLE
Spectator HUD enhancements

### LINKED ISSUE #1074: toggle_jump Does Not Swap Perspective While Spectating
**Describe the bug**
A key bound to `toggle_jump` does not swap perspective (1st person to 3rd person) of the person you are spectating, while `+jump` keys do, which is unintuitive.

**To Reproduce**
1. Join a map and spectate a player
2. Bind a key to toggle jump (in settings or console)
3. They key does not

**Expected behavior**
`toggle_jump` and `+jump` should both function the same exact way while spectating

**Desktop/Branch (please complete the following information):**
 - OS: Windows
 - Branch: Steam version

**Additional context**
This is probably not technically a bug but it feels very unintuitive for "the jump key" to sometimes do what it is meant to do while spectating and sometimes not.

### LINKED ISSUE #1076: Cursor sometimes cannot interact with spectator hud while chat box is open
**What feature is your improvement idea related to? Please describe.**
When using chat while spectating, your cursor can sometimes interact with the hud (arrows to change target, X to stop spectating) and sometimes not, depending on if you had cursor input enabled before opening chat with the DUCK key.

video: https://streamable.com/1mzbod
In the video my cursor is invisible but you can see the boxes over the buttons when I hover over them and they are clickable.

**Describe the solution you'd like**
The cursor should always remain functional (or nonfunctional) when it is visible for consistency. 

**Describe alternatives you've considered, if any**
This is not necessary and could easily be ignored and left as is. Alternatively, the cursor could never interact with the spectator hud while chat is open but that is most likely less user friendly.

**Additional context**
When opening chat while the cursor is open, it teleports from wherever it is to the center of the chat box. This is a somewhat related issue but seemed too small to open another card, it can be split off if desired. Ideally the cursor doesn't teleport around and stays in place when it is already visible.

### LINKED ISSUE #1088: Add freecam spectate mode
**What feature is your improvement idea related to? Please describe.**
It is possible to toggle between 1st and 3rd person while spectating another player, but it is impossible to navigate into freecam mode without using the console.

**Describe the solution you'd like**
Using jump should swap between 1st person, 3rd person, and a freecam view. 
Alternatively, there could be a button added to the hud (similar to the X 'stop spectating' button) that goes between freecam and the current view.

**Additional context**
It is possible to enter freecam by typing `spectate` in console, but it has no autocomplete and outputs an error line saying `CBasePlayer::ChangeTeam( 1 ) - invalid team index.`, meaning it is most likely not intended. 

This issue is pretty close to another small spectator-related issue #1074

### PR DESCRIPTION
**Pull request by braem**
Closes #1088 
Closes #1074 
Closes #1076 

A few minor spectating enhancements & fixes. Figured I'd do this now that there are a decent amount of lobbies to test in.

- Allow roaming mode in spectate. Done like in other source games (first person -> third person -> free roam).
- Allows toggle_jump to change spectate mode and toggle_duck to enable spectator mouse input.
- Enable Spectator HUD mouse input when chat is opened.
  - Not too sure about this one as it's doing a dynamic cast. Would have to change the spec hud to use a global instead of being a viewport panel to fix i believe.

https://cdn.discordapp.com/attachments/439686613036171265/770435396726685696/2020-10-26_13-17-18.mp4

### Checklist
- [x] **I have thoroughly tested all of the code I have modified/added/removed to ensure something else did not break**
- [x] If there is a localization token change, I have updated the `momentum_english_ref.res` file with the changes, ran `tokenizer.py` to generate an up-to-date localization file, and have committed both the `.res` file changes and the new localization `.txt` file
- [x] If I introduced new h/cpp files, I have added them to the appropriate project's VPC file (`server_momentum.vpc` / `client_momentum.vpc` / etc)
- [x] If I have added or modified any visual assets (models, materials, panels, effects, etc), I have taken screenshots / videos of them and attached them to this PR directly (screenshots uploaded through github, videos uploaded to youtube and linked)
- [x] If I have modified any console command, console variable, or momentum entity, I have opened an issue (or a PR) for it in the [Momentum Mod documentation repository](https://github.com/momentum-mod/docs)
- [x] My commits are relatively small and scoped to the best of my ability
- [x] My branch has a clear history of changes that can be easy to follow when being reviewed commit-by-commit
- [x] My branch is functionally complete; the only changes to be done will be those potentially requested in code review

### CODE REVIEW

File: mp/src/game/client/hud_chat.cpp [hunk 1]
```diff
@@ -53,6 +54,9 @@ void CHudChat::StartMessageMode()
     SetMouseInputEnabled(true);
     SetKeyBoardInputEnabled(true);
 
+    const auto specGUI = dynamic_cast<CMOMSpectatorGUI *>(gViewPortInterface->FindPanelByName(PANEL_SPECGUI));
```
Comment (Gocnak) [inline mp/src/game/client/hud_chat.cpp:57]: I think this is backwards; the chat panel shouldn't care about other particular panels.

The spec GUI on the other hand explicitly cares when chat or the map finished panels are open and have mouse input. This kind of logic should instead be there.

**Review by Gocnak — CHANGES_REQUESTED** [review]

File: mp/src/game/client/hud_chat.cpp (hunk 1 above)
Comment (braem) [reply mp/src/game/client/hud_chat.cpp]: Did this that way using an event.

File: mp/src/game/client/momentum/ui/spectate/mom_spectator_gui.cpp [hunk 2]
```diff
@@ -140,14 +142,19 @@ void CMOMSpectatorGUI::OnMousePressed(MouseCode code)
...
+    if (FStrEq(pEvent->GetName(), "spec_target_updated"))
     {
         // So apparently calling Update from here doesn't work, due to some weird
         // thing that happens upon the player's m_hObserverTarget getting updated.
         // Pushing this back three ticks is more than long enough to delay the Update()
         // to fill the panel with the replay's info.
         m_flNextUpdateTime = gpGlobals->curtime + gpGlobals->interval_per_tick * 3.0f;
     }
+
+    if (FStrEq(pEvent->GetName(), "hud_chat_opened"))
```
Comment (Gocnak) [inline mp/src/game/client/momentum/ui/spectate/mom_spectator_gui.cpp:154]: Should do else if here.

File: mp/src/game/client/momentum/clientmode_mom_normal.cpp [hunk 3]
```diff
@@ -208,37 +208,43 @@ int ClientModeMOMNormal::HudElementKeyInput(int down, ButtonCode_t keynum, const
...
-            // MOM_TODO: re-enable this in alpha+ when we add movie-style controls to the spectator menu!
-            // m_pViewport->ShowPanel(PANEL_SPECMENU, true);
-
-            return 0; // we handled it, don't handle twice or send to server
-        }
+        bool bMouseState = m_pSpectatorGUI->IsMouseInputEnabled();
+        m_pSpectatorGUI->SetMouseInputEnabled(!bMouseState);
+        if (m_pHudMapFinished && m_pHudMapFinished->IsVisible())
+            m_pHudMapFinished->SetMouseInputEnabled(!bMouseState);
+        // MOM_TODO: re-enable this in alpha+ when we add movie-style controls to the spectator menu!
```
Comment (Gocnak) [inline mp/src/game/client/momentum/clientmode_mom_normal.cpp:224]: Remove this todo, we have replay controls UI for this

**Review by Gocnak — CHANGES_REQUESTED** [review]

**Review by Gocnak — APPROVED** [review]

**Pull request merged**